cns/restserver/restserver.go

```diff
@@ -1252,7 +1263,31 @@ func (service *HTTPRestService) getNetworkContainerResponse(req cns.GetNetworkCo
 		}
 
 		logger.Printf("pod info %+v", podInfo)
-		containerID = service.state.ContainerIDByOrchestratorContext[podInfo.PodName+podInfo.PodNamespace]
+
+		var (
+			context      = podInfo.PodName + podInfo.PodNamespace
+			dncEP        = service.GetOption(acn.OptPrivateEndpoint).(string)
+			infraVnet    = service.GetOption(acn.OptInfrastructureNetwork).(string)
+			nodeID       = service.GetOption(acn.OptNodeID).(string)
+			isManagedDnc = dncEP != "" && infraVnet != "" && nodeID != ""
```

Comment:
Consider a case where partner team starts CNS without specifying infraVnet- We are not throwing error and because of logic on line 1272, we are not going to do things needed for managed dnc. This would go undetected and CNS won't get any data for that NC.
In CNS is used with mdnc, we need to validate these params at cns start and throw and error.

Reply:
I agree 100%. Instead of "calculating" that we are in managed dnc mode, pass in an explicit argument that it is managed DNC mode and then validate all the necessary arugments that are required for that mode.